test/test_nn.py

```diff
@@ -12067,6 +12067,76 @@ def test_threshold_inplace_overlap(self, device):
         F.threshold(x, 0.5, 0.5, inplace=True)
         F.threshold_(x, 0.5, 0.5)
 
+    def test_triplet_margin_loss_with_distance_default_parity(self, device):
+        # Test for `nn.TripletMarginLossWithDistance` and
+        # `F.triplet_margin_loss_with_distance`.  Checks
+        # for parity against the respective non-distance-agnostic
+        # implementations of triplet margin loss (``nn.TripletMarginLoss`
+        # and `F.triplet_margin_loss`) under *default args*.
+
+        anchor = torch.randn(5, 10, device=device, requires_grad=True)
+        positive = torch.randn(5, 10, device=device, requires_grad=True)
+        negative = torch.randn(5, 10, device=device, requires_grad=True)
+
+        # functional grad and parity check
+        self.assertTrue(gradcheck(lambda a, p, n: F.triplet_margin_loss_with_distance(
+            a, p, n), (anchor, positive, negative)))
+        self.assertEqual(F.triplet_margin_loss_with_distance(anchor, positive, negative),
+                         F.triplet_margin_loss(anchor, positive, negative))
+
+        # module grad and parity check
+        loss_base = nn.TripletMarginLoss()
+        loss_test = nn.TripletMarginLossWithDistance()
+        self.assertTrue(gradcheck(lambda a, p, n: loss_test(
+            a, p, n), (anchor, positive, negative)))
+        self.assertEqual(loss_test(anchor, positive, negative),
+                         loss_base(anchor, positive, negative))
+
+    def test_triplet_margin_loss_with_distance(self, device):
+        # Test for `nn.TripletMarginLossWithDistance` and
```

Comment:
I understand what this test is trying to accomplish, and it makes sense, but I don't think it's necessary. Instead, how about iterating over a variety of distance functions, running gradcheck, and comparing the functional and module implementations of triplet_margin_loss_with_distance?

Additionally, you can compare the functional version, the module version, and a traced (and scripted?) version of the module. Just trace in advance and then you can iterate over the ops and ensure their results are all equal to each other.